src/maxminddb.c

```diff
@@ -1116,13 +1119,13 @@ int MMDB_get_value(MMDB_entry_s *const start,
 int MMDB_vget_value(MMDB_entry_s *const start,
                     MMDB_entry_data_s *const entry_data,
                     va_list va_path) {
-    int length = path_length(va_path);
+    size_t length = path_length(va_path);
     const char *path_elem;
     int i = 0;
 
-    MAYBE_CHECK_SIZE_OVERFLOW(length,
-                              SIZE_MAX / sizeof(const char *) - 1,
-                              MMDB_INVALID_METADATA_ERROR);
+    if (length == SIZE_MAX) {
```

Comment:
I'm actually unsure if this conditional is necessary anymore

Reply:
I was thinking it would be technically possible to have a va_list that had exactly this many arguments. Then when we increment it by one below, we'd wrap around and the argument to calloc wouldn't be as we'd expect. Do you think that'd not be possible for some reason?

Reply:
My thought was size_t would be big enough to index the string including a terminating NULL, such that length would never be as long as the supplies string. I haven’t tested that theory though, and don’t see how the conditional would ever do something unwanted so I was just curious.

Reply:
It's a count of the number of elements in the `va_list` rather than a string. I haven't found anything saying what the maximum number of elements in a `va_list` is, which could decide this. What you say is possibly true though. I think it's probably safest to leave it unless we can know for sure that a `va_list` couldn't have exactly this many elements in any implementation. Possibly I'm missing something though!